entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlementApi.java

```diff
@@ -156,6 +157,58 @@ public Entitlement doCall(final EntitlementApi entitlementApi, final Entitlement
     }
 
     @Override
+    public Entitlement createBaseEntitlementWithAddOns(final UUID accountId, final Iterable<EntitlementSpecifier> entitlementSpecifier, final LocalDate effectiveDate,
+                                                             final Iterable<PluginProperty> properties, final CallContext callContext) throws EntitlementApiException {
+
+        final EntitlementSpecifier baseSpecifier = Iterables.tryFind(entitlementSpecifier, new Predicate<EntitlementSpecifier>() {
+            @Override
+            public boolean apply(final EntitlementSpecifier specifier) {
+                return specifier.getPlanPhaseSpecifier() != null && ProductCategory.BASE.equals(specifier.getPlanPhaseSpecifier().getProductCategory());
+            }
+        }).orNull();
```

Comment:
We could rely on convention from top to bottom of the stack to always pass the BASE first in the list (and just do a check at each level to verify the first entry is for the BASE).